src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/SubmissionComponent.scala

```diff
@@ -69,7 +74,212 @@ trait SubmissionComponent {
+          throw new RawlsException(s"A submission already exists by the id [${submission.submissionId}]")
+      }
+    } map { _ => submission }
+
+    /* updates the status of a submission */
+    def updateStatus(workspaceContext: SlickWorkspaceContext, submissionId: String, newStatus: SubmissionStatuses.SubmissionStatus) = {
+      findById(UUID.fromString(submissionId)).map(_.status).update(newStatus.toString)
+    }
+
+    /* deletes a submission and all associated records */
+    def delete(workspaceContext: SlickWorkspaceContext, submissionId: String): ReadWriteAction[Boolean] = {
+
+      def deleteSubmissionWorkflows(submissionId: UUID) = {
+        workflowQuery.findWorkflowsBySubmissionId(submissionId).result.flatMap { recs => DBIO.sequence(recs.map { rec =>
+          workflowQuery.delete(workspaceContext, submissionId.toString, rec.id.toString)
+        })}
+      }
+
+      uniqueResult[SubmissionRecord](findById(UUID.fromString(submissionId))) flatMap {
+        case None =>
+          DBIO.successful(false)
+        case Some(submissionRec) => {
+          deleteSubmissionWorkflows(UUID.fromString(submissionId)) andThen
+            workflowQuery.findInactiveWorkflows(UUID.fromString(submissionId)).delete andThen
+            findById(UUID.fromString(submissionId)).delete
+        } map { count =>
+          count > 0
+        }
+      }
+    }
+
+    /* admin action: lists all active submissions in the system */
+    def listAllActiveSubmissions(): ReadAction[Seq[ActiveSubmission]] = {
+      findActiveSubmissions.result.flatMap(recs => DBIO.sequence(recs.map{ rec =>
+        loadActiveSubmission(rec.id)
+      }))
+    }
+
+    /*
+      the finder methods
+     */
+
+    def findByWorkspaceId(workspaceId: UUID): SubmissionQueryType = {
+      filter(rec => rec.workspaceId === workspaceId)
+    }
+
+    def findById(submissionId: UUID): SubmissionQueryType = {
+      filter(rec => rec.id === submissionId)
+    }
+
+    def findActiveSubmissions: SubmissionQueryType = {
+      filter(rec => rec.status inSetBind(Seq(SubmissionStatuses.Aborting.toString, SubmissionStatuses.Submitted.toString)))
+    }
+
+    /*
+      the load methods
+     */
+
+    def loadSubmission(submissionId: UUID): ReadAction[Option[Submission]] = {
+      uniqueResult[SubmissionRecord](findById(submissionId)) flatMap {
+        case None => DBIO.successful(None)
+        case Some(submissionRec) =>
+          for {
+            config <- methodConfigurationQuery.loadMethodConfigurationById(submissionRec.methodConfigurationId)
+            workflows <- loadSubmissionWorkflows(submissionRec.id)
+            entity <- loadSubmissionEntity(submissionRec.submissionEntityId)
+            notStarted <- workflowQuery.loadInactiveWorkflows(submissionRec.id)
+          } yield Option(unmarshalSubmission(submissionRec, config.get, entity, workflows, notStarted))
+      }
+    }
+
+    def loadActiveSubmission(submissionId: UUID): ReadAction[ActiveSubmission] = {
```

Comment:
I didn't have this as an option because the submissionId was supplied by a query on all active submissions, so we already know that the submission actually exists.